File: matchingService/src/QueueService/MatchRequest.ts
class MatchRequest {
    private userId: string;
    private topic: string;
    private difficulty: string;

    constructor(userId: string, topic: string, difficulty: string) {
        this.userId = userId;
        this.topic = topic;
        this.difficulty = difficulty;
    }

    public getTopic(): string {
        return this.topic;
    }

    public getDifficulty(): string {
        return this.difficulty;
    }
}

export default MatchRequest;

File: backend/matching/src/main/java/com/example/backend/matching/model/MatchRequest.java
package com.example.backend.matching.model;

public class MatchRequest {
    private String userId;
    private String programmingLanguage;
    private String difficulty;
    private String topics;

    // Getters and setters
    public String getUserId() { return userId; }
    public void setUserId(String userId) { this.userId = userId; }

    public String getProgrammingLanguage() { return programmingLanguage; }
    public void setProgrammingLanguage(String programmingLanguage) { this.programmingLanguage = programmingLanguage; }

    public String getDifficulty() { return difficulty; }
    public void setDifficulty(String difficulty) { this.difficulty = difficulty; }

    public String getTopics() { return topics; }
    public void setTopics(String topics) { this.topics = topics; }
}
